Migrate/handle old style loadout data

A user has their loadout data saved in the old format where the loadout has no key value for the name and any time they use loadout show they can't get info about their saved loadout.

```
[2020-06-22 16:51:27] [ERROR] red: Exception in command 'loadout show'
Traceback (most recent call last):
  File "/.../cat38/lib/python3.8/site-packages/discord/ext/commands/core.py", line 83, in wrapped
    ret = await coro(*args, **kwargs)
  File "/.../Red-DiscordBot/cogs/CogManager/cogs/adventure/adventure.py", line 1644, in show_loadout
    stats = await self._build_loadout_display({"items": loadout})
  File "/.../Red-DiscordBot/cogs/CogManager/cogs/adventure/adventure.py", line 4228, in _build_loadout_display
    item = Item.from_json(data)
  File "/.../Red-DiscordBot/cogs/CogManager/cogs/adventure/charsheet.py", line 211, in from_json
    data = data[name]
KeyError: 'slotattchararitydexluckowned'
```

--- a/adventure/adventure.py
+++ b/adventure/adventure.py
@@ -66,7 +66,7 @@ log = logging.getLogger("red.cogs.adventure")
 
 REBIRTH_LVL = 20
 REBIRTH_STEP = 10
-_SCHEMA_VERSION = 2
+_SCHEMA_VERSION = 3
 _config: Config = None
 
 
@@ -214,7 +214,7 @@ class AdventureResults:
 class Adventure(BaseCog):
     """Adventure, derived from the Goblins Adventure cog by locastan."""
 
-    __version__ = "3.2.24"
+    __version__ = "3.2.25"
 
     def __init__(self, bot: Red):
         self.bot = bot
@@ -540,7 +540,38 @@ class Adventure(BaseCog):
                         adventurers_data[user]["loadouts"] = new_loadout
                     except Exception:
                         adventurers_data[user]["loadouts"] = {}
-        await self.config.schema_version.set(to_version)
+            await self.config.schema_version.set(2)
+            from_version = 2
+        if from_version < 3 <= to_version:
+            group = self.config._get_base_group(self.config.USER)
+            accounts = await group.all()
+            tmp = accounts.copy()
+            async with group.all() as adventurers_data:
+                for user in tmp:
+                    new_loadout = {}
+                    if "loadouts" not in adventurers_data[user]:
+                        adventurers_data[user]["loadouts"] = {}
+                    try:
+                        for (loadout_name, loadout) in adventurers_data[user]["loadouts"].items():
+                            if loadout_name in {
+                                "head",
+                                "neck",
+                                "chest",
+                                "gloves",
+                                "belt",
+                                "legs",
+                                "boots",
+                                "left",
+                                "right",
+                                "ring",
+                                "charm",
+                            }:
+                                continue
+                            new_loadout[loadout_name] = loadout
+                        adventurers_data[user]["loadouts"] = new_loadout
+                    except Exception:
+                        adventurers_data[user]["loadouts"] = {}
+            await self.config.schema_version.set(3)
 
     def _convert_item_migration(self, item_name, item_dict):
         new_name = item_name
@@ -6295,10 +6326,8 @@ class Adventure(BaseCog):
                 c.skill["pool"] += ending_points - starting_points
                 if c.skill["pool"] > 0:
                     extra = _(" You have **{}** skill points available.").format(c.skill["pool"])
-                rebirth_text = (
-                    _("{} {} is now level **{}**!{}\n{}").format(
-                        levelup_emoji, user.mention, lvl_end, extra, rebirthextra
-                    ),
+                rebirth_text = _("{} {} is now level **{}**!{}\n{}").format(
+                    levelup_emoji, user.mention, lvl_end, extra, rebirthextra
                 )
             if c.rebirths > 1:
                 roll = random.randint(1, 100)

